hoomd/hpmc/ComputeSDF.h

```diff
@@ -305,7 +335,7 @@ template<class Shape> Scalar ComputeSDF<Shape>::getMaxInteractionDiameter()
 */
 template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
     {
-    if (m_mc->getPatchEnergy())
+    if (m_mc->getPatchEnergy() || m_mc->requiresExpansiveSDF())
```

Comment:
There is no need to introduce a new method in every shape. Use C++ constexpr to determine the expansive flag based on the type of `Shape`. For safety, default to `true` and only set to `false` for known convex `Shape` classes.